src/main/java/amazin/controller/BookController.java

```diff
@@ -0,0 +1,82 @@
+package amazin.controller;
+
+import java.util.Optional;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.SessionAttributes;
+
+import amazin.model.Book;
+import amazin.service.BookService;
+
+@Controller
+@SessionAttributes("book")
+public class BookController {
+    public static final String VIEW_CREATE_BOOK = "create-book";
+    public static final String VIEW_UPDATE_BOOK = "update-book";
+    public static final Object MODEL_ATTRIBUTE_BOOK = "book";
+
+    @Autowired
+    private final BookService bookService;
+
+    public BookController(final BookService bookService) {
+        this.bookService = bookService;
+    }
+
+    @GetMapping("/addbook")
+    public String addBookForm(Model model) {
+        model.addAttribute("book", new Book());
+        return VIEW_CREATE_BOOK;
+    }
+
+    @PostMapping("/addbook")
+    public String addBook(@Valid @ModelAttribute Book book, BindingResult result, Model model) {
+        if (result.hasErrors()) {
+            return VIEW_CREATE_BOOK;
+        }
+
+        bookService.create(book);
+        model.addAttribute("books", bookService.getAll());
```

Comment:
won't this be overridden when redirecting "/" gets called?

Reply:
You are absolutely correct, this is not necessary as it will not be used.